test/parallel/test-http2-client-port-80.js

```diff
@@ -0,0 +1,16 @@
+'use strict';
+
+const common = require('../common');
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+const assert = require('assert');
+const http2 = require('http2');
+const net = require('net');
+
+// Verifies that port 80 gets set as expected
+
+net.connect = common.mustCall((port) => {
+  assert.strictEqual(port, '80');
+  process.exit();
```

Comment:
Instead of calling `process.exit()` here, can you let the test exit naturally.

Reply:
Then I need to do a bunch of error handling for the *about to fail* `http2.connect` which seemed worse to me because then it becomes super reliant on the exact code within that function (or I need to wrap it in `assert.throws` and that also seemed super sketchy).

Reply:
I definitely wouldn't call it "super sketchy" but I get what you mean. Can you try to keep that in mind for the future though. We generally try to avoid `process.exit()` in tests.

Reply:
I feel like any solution I come up with here is a bit sketchy haha. I don't love `process.exit` but it's less convoluted than the other alternative I can immediately come up with, which is:

> I suppose I could throw a custom error within net.connect and then catch that outside.

Reply:
@apapirovski how about something like this:

```js
const original = net.connect;

net.connect = common.mustCall((...args) => {
  assert.strictEqual(args[0], '80');
  return original(...args);
});

// Use a non-routable IP address.
const client = http2.connect('http://192.0.2.1:80');
client.destroy();
```